### PR TITLE
Added write check during initial scan

### PR DESCRIPTION
**Pull request by snar5**
This will check for write access during initial scan and note each as it scans.

### CODE REVIEW

**Comment by bcook-r7** [comment]
Are you sure that you need to write to a share to actually check if it is writable? If we can check without writing, then this is something that does not even need to be an option.

**Comment by snar5** [comment]
I changed it so that on my initial scans I could see what access the user had, noting especially those that had an admin share writable. As it stands now it might list C$ as a share but then if you try smbclient or something similar you get permission denied. So this option allows me to see on the first attempt what I can write to as this selected user.  


     On Monday, March 16, 2015 1:47 PM, Brent Cook <notifications@github.com> wrote:
   

 Are you sure that you need to write to a share to actually check if it is writable? If we can check without writing, then this is something that does not even need to be an option.—

File: modules/auxiliary/scanner/smb/smb_enumshares.rb [hunk 1]
```diff
@@ -274,10 +280,16 @@ def srvsvc_netshareenum(ip)
       name    = Rex::Text.to_ascii(name)
       s_type  = Rex::Text.to_ascii(share_type(types[t]))
       comment = Rex::Text.to_ascii(comment)
-
+      if datastore['Check_Write_Access']
```
Comment (jvazquez-r7) [inline modules/auxiliary/scanner/smb/smb_enumshares.rb]: You're checking write access only when using SRVSVC, which is unexpected I think.

**Comment by jvazquez-r7** [comment]
@snar5 you're right, what @bcook-r7 is pointing is which you're writing and deleting a file in order to check write access. Which is kinda noisy, specially for the file system activity.

I would bet you can check write access with a QUERY_PATH_INFORMATION transaction (https://msdn.microsoft.com/en-us/library/ee441924.aspx). Ask for SMB_INFO_STANDARD information level (https://msdn.microsoft.com/en-us/library/ff470155.aspx). The response should return the SMB_FILE_ATTRIBUTES for the path: https://msdn.microsoft.com/en-us/library/ee441551.aspx. The attributes include if the path has the  SMB_FILE_ATTRIBUTE_READONLY or not. Isn't it good enough here? In this way you wouldn't write and deleting from file system, making the check more convenient I think.

**Comment by jvazquez-r7** [comment]
On the other hand, the module should be msftidy compliant! Thanks!

**Pull request closed without merging**